wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/vfs/impl/file/event/VfsEventService.java

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.vfs.impl.file.event;
+
+import com.google.common.annotations.Beta;
+
+import org.slf4j.Logger;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static java.util.concurrent.Executors.newSingleThreadExecutor;
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * @author Dmitry Kuleshov
+ *
+ * @since 4.5
+ */
+@Beta
+abstract class VfsEventService {
+    private static final Logger LOG = getLogger(VfsEventService.class);
+
+    private final ExecutorService executor;
+    private final AtomicBoolean   running;
+
+    public VfsEventService() {
+        this.executor = newSingleThreadExecutor();
```

Comment:
Please use thread builder factory to set name of the thread. That will help in case we will need to analyze thread dump. Here is an example of the way we usually use it https://github.com/eclipse/che/blob/25e6cc2360133ead8847feccf1f36a5c3f03a2fb/wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java#L110-L110